docs: complete the target configuration (#7168)
web-infra-dev/rspack: website/docs/en/config/target.mdx

| options                      | description                                                                                                                                                                                                                                   |
| ---------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `'web'`                      | Compile as available in the browser environment (default)                                                                                                                                                                                     |

> | `'webworker'`                | Compile as WebWorker                                                                                                                                                                                                                          |

Reviewer: `web worker`?

> | `'webworker'`                | Compile as WebWorker                                                                                                                                                                                                                          |

Author: ```suggestion
| `'webworker'`                | Compile as Web Worker                                                                                                                                                                                                                          |
```

> | `'browserslist'`             | Infer the ECMAScript version based on the configured browserslist (default if browserslist config is available)                                                                                                                               |

Reviewer: ```suggestion
| `'browserslist'`             | Infer the ES-features version based on the configured browserslist (default if browserslist config is available)                                                                                                                               |
```

The copywriting on webpack doc is

> Infer a platform and the ES-features

Rspack / webpack will try to map browserslist direct to ES-feature https://github.com/web-infra-dev/rspack/blob/main/packages/rspack/src/config/target.js#L24 instead of a certain ECMA version.

};
```

When multiple targets are passed, then common subset of features will be used:

```ts title="rspack.config.js"
module.exports = {
  // Rspack will generate a runtime code for web platform and will use only ES5 features
  target: ['web', 'es5'],
};
```

Note that not all targets may be mixed for now. When specifying that the Compiler needs to be compiled for multiple platforms, an error is reported:

```ts title="rspack.config.js"
module.exports = {
  target: ['web', 'node'],
};
```

For this case, you can define multiple Rspack configurations for bundling based on [MultiCompiler](/api/javascript-api#multicompiler).

## browserslist

If a project has a browserslist config, then Rspack will use it for:

- Determinate ES-features that may be used to generate a runtime-code.
- Infer an environment (e.g: `last 2 node versions` the same as `target: "node"` with some [`output.environment`](/config/output#outputenvironment) settings).

:::tip What is Browserslist

> [Browserslist](https://browsersl.ist/) can specify which browsers your web application can run in, it provides a configuration for specifying browsers range. Browserslist has become a standard in the industry, it is used by libraries such as Autoprefixer, Babel, ESLint, PostCSS, SWC and Webpack.

Reviewer: ```suggestion
[Browserslist](https://browsersl.ist/) can specify which browsers your web application can run in, it provides a configuration for specifying browsers range. Browserslist has become a standard in the industry, it is used by libraries such as Autoprefixer, Babel, ESLint, PostCSS, SWC and webpack.
```
